sbin/build.sh

```diff
@@ -931,51 +931,121 @@ setPlistValueForMacOS() {
   fi
 }
 
-# Get the tags from the git repo and choose the latest chronologically ordered tag for the given JDK version.
+# JDK8 tag selection
+# Version tags are expected to follow this pattern:
+#   jdk8uV-bB
+# where V and B are numeric.
+# - answer the largest matching tag based on the version ordering of V.B
 #
-# Note, we have to chronologically order, as with a Shallow cloned (depth=1) git repo there is no "topo-order"
-# for tags, also commit date order cannot be used either as the commit dates do not necessarily follow chronologically.
+getLatestTagJDK8() {
+  local max_tag=
+  local max_v=
+  local max_b=
+
+  while read -r tag ; do
+    if [[ "$tag" =~ ^jdk8u([0-9]+)-b([0-9])+$ ]] ; then
+      local better=0
+      local cur_v=${BASH_REMATCH[1]}
+      local cur_b=${BASH_REMATCH[2]}
+
+      if [ -z "$max_tag" ] ; then
+        better=1
+      elif [ "$cur_v" -gt "$max_v" ] ; then
+        better=1
+      elif [ "$cur_v" -eq "$max_v" ] && [ "$cur_b" -gt "$max_b" ] ; then
+        better=1
+      fi
+
+      if [ "$better" -ne 0 ] ; then
+        max_tag="$tag"
+        max_v="$cur_v"
+        max_b="$cur_b"
+      fi
+    fi
+  done
+
+  if [ -n "$max_tag" ] ; then
+    echo "$max_tag"
+  fi
+}
+
+# JDK11+ tag selection
+# Version tags are expected to follow this pattern:
+#   jdk-V[.W[.X[.P]]]+B
+# where V, W, X, P and B are numeric (missing components are considered to be 0).
+# - answer the largest matching tag based on the version ordering of V.W.X.P+B
+#
+getLatestTagJDK11plus() {
+  local max_tag=
+  local max_v=
+  local max_w=
+  local max_x=
+  local max_p=
+  local max_b=
+
+  while read -r tag ; do
+    if [[ "$tag" =~ ^jdk-([0-9]+)(\.([0-9]+))?(\.([0-9]+))?(\.([0-9]+))?\+([0-9])+$ ]] ; then
```

Comment:
Same here for build group? ([0-9])+ => ([0-9]+)

Reply:
Fixed. Thanks, Andrew.